giraph-core/src/main/java/org/apache/giraph/writable/kryo/HadoopKryo.java

```diff
@@ -111,19 +123,15 @@ public Kryo create() {
         "Logger must be a static field");
   }
 
-  // Use chunked streams, so within same stream we can use both kryo and
-  // non-kryo serialization.
   /** Reusable Input object */
-  private final InputChunked input = new InputChunked(4096);
+  private InputChunked input;
```

Comment:
Should we separate two Kryo implementations in two classes since they have a lot of differences?

Reply:
Agreed, we can refactor the existing code to have a better code readability. I have added a TODO for that.